systemtest/src/test/java/io/strimzi/systemtest/kafka/listeners/ListenersST.java

```diff
@@ -2006,6 +2010,112 @@ void testCertificateWithNonExistingDataKey(ExtensionContext extensionContext) {
+
+        KafkaUser kafkaUser = KafkaUserTemplates.scramShaUser(clusterName, userName)
+            .editSpec()
+                .withNewKafkaUserScramSha512ClientAuthentication()
+                    .withNewPassword()
+                        .withNewValueFrom()
+                            .withNewSecretKeyRef("password", secretName, false)
+                        .endValueFrom()
+                    .endPassword()
+                .endKafkaUserScramSha512ClientAuthentication()
+            .endSpec()
+            .build();
+
+        resourceManager.createResource(extensionContext, KafkaTemplates.kafkaEphemeral(clusterName, 3)
+            .editSpec()
+                .editKafka()
+                    .withListeners(new GenericKafkaListenerBuilder()
+                        .withType(KafkaListenerType.INTERNAL)
+                        .withName(Constants.TLS_LISTENER_DEFAULT_NAME)
+                        .withPort(9096)
+                        .withTls(true)
+                        .withNewKafkaListenerAuthenticationScramSha512Auth()
+                        .endKafkaListenerAuthenticationScramSha512Auth()
+                        .build())
+                .endKafka()
+            .endSpec()
+            .build(),
+            kafkaUser,
+            KafkaTopicTemplates.topic(clusterName, topicName).build()
+        );
+
+        resourceManager.createResource(extensionContext, KafkaClientsTemplates.kafkaClients(namespaceName, true, kafkaClientsName, kafkaUser).build());
+
+        String kafkaClientsPodName =
+            kubeClient(namespaceName).listPodsByPrefixInName(namespaceName, kafkaClientsName).get(0).getMetadata().getName();
+
+        InternalKafkaClient internalKafkaClient = new InternalKafkaClient.Builder()
+            .withUsingPodName(kafkaClientsPodName)
+            .withTopicName(topicName)
+            .withNamespaceName(namespaceName)
+            .withClusterName(clusterName)
+            .withKafkaUsername(userName)
+            .withMessageCount(MESSAGE_COUNT)
+            .withListenerName(Constants.TLS_LISTENER_DEFAULT_NAME)
+            .build();
+
+        int sentMessages = internalKafkaClient.sendMessagesTls();
+        internalKafkaClient.checkProducedAndConsumedMessages(
+            sentMessages,
+            internalKafkaClient.receiveMessagesTls()
+        );
+
+        LOGGER.info("Changing password in secret: {}, we should be able to send/receive messages", secretName);
+
+        password = new SecretBuilder(password)
+            .addToData("password", secondEncodedPassword)
+            .build();
+
+        kubeClient().namespace(namespaceName).createSecret(password);
+        assertThat("Password in secret is not correct", kubeClient().namespace(namespaceName).getSecret(secretName).getData().get("password"), is(secondEncodedPassword));
+
+        LOGGER.info("We need to recreate Kafka Clients deployment, so the correct password from secret will be taken");
```

Comment:
I think you might need some wait before this? The password is changed only in next periodical reconciliation, not immediately when you change the secret.

Reply:
Okay, is there any way how to check that password is already changed? Or should I just wait some time and then continue with the test?

Reply:
I guess you can for example watch for when the regular user secret changes

Reply:
That sounds good, I'll add the wait then, thanks! :)